sdks/go/pkg/beam/io/synthetic/source.go

```diff
@@ -262,8 +281,10 @@ func (b *SourceConfigBuilder) BuildFromJSON(jsonData []byte) SourceConfig {
 // synthetic source. It should be created via a SourceConfigBuilder, not by
 // directly initializing it (the fields are public to allow encoding).
 type SourceConfig struct {
-	NumElements   int `json:"num_records"`
-	InitialSplits int `json:"initial_splits"`
-	KeySize       int `json:"key_size"`
-	ValueSize     int `json:"value_size"`
+	NumElements    int `json:"num_records"`
+	InitialSplits  int `json:"initial_splits"`
+	HotKeys        int `json:"hot_keys"`
```

Comment:
Python tests use `num_hot_keys`, let's do the same in Go. `hot_key_fraction` is correct.

Reply:
I think you forgot about setters for your new variables. Please add them, because right now, we can only set them via JSON.

Reply:
```
	NumHotKeys     int     `json:"num_hot_keys"`
	HotKeyFraction float64 `json:"hot_key_fraction"`
```

Reply:
```
// NumHotKeys determines the number of keys with the same value among
// generated keys.
//
// Valid values are in the range of [0, ...] and the default value is 0.
func (b *SourceConfigBuilder) NumHotKeys(val int) *SourceConfigBuilder {
	b.cfg.NumHotKeys = val
	return b
}

// HotKeyFraction determines the value of hot key fraction.
//
// Valid values are floating point numbers from 0 to 1.
func (b *SourceConfigBuilder) HotKeyFraction(val float64) *SourceConfigBuilder {
	b.cfg.HotKeyFraction = val
	return b
}
```